lib/services.template.ejs

```diff
@@ -1,87 +1,110 @@
 (function(window, angular, undefined) {'use strict';
 
-var urlBase = <%-: urlBase | q %>;
 var authHeader = 'authorization';
+var _urlBase = <%-: urlBase | q %>;
+var urlBase = _urlBase;
 
 /**
  * @ngdoc overview
- * @name <%- moduleName %>
+ * @name <%-: registryName %>
  * @module
  * @description
  *
- * The `<%- moduleName %>` module provides services for interacting with
- * the models exposed by the LoopBack server via the REST API.
+ * The `<%- registryName %>`.registry module provides services for interacting
+ * with multiple running instances of the same REST API.
  *
  */
-var module = angular.module(<%-: moduleName | q %>, ['ngResource']);
+var module = angular.module(<%-: registryName | q %>, ['ngResource']);
 
-<% for (var modelName in models) {
-     var meta = models[modelName];
-
-     // capitalize the model name
-     modelName = modelName[0].toUpperCase() + modelName.slice(1);
--%>
+<% //TODO: docs -%>
 /**
  * @ngdoc object
- * @name <%-: moduleName %>.<%-: modelName %>
- * @header <%-: moduleName %>.<%-: modelName %>
+ * @name <%-: registryName %>.Loopback
+ * @header <%-: registryName %>.Loopback
  * @object
  *
  * @description
  *
- * A $resource object for interacting with the `<%-: modelName %>` model.
- *
- * ## Example
- *
- * See
- * {@link http://docs.angularjs.org/api/ngResource.$resource#example $resource}
- * for an example of using this object.
- *
+ * Object that gives you access to the LoopBack models
 <% /*
     * TODO(bajtos) provide an example of performing basic CrUD operations,
     * including filter/where find query
     */ -%>
  */
 module.factory(
-  <%-: modelName | q %>,
+  <%-: registryName | q %>,
   ['LoopBackResource', 'LoopBackAuth', '$injector', function(Resource, LoopBackAuth, $injector) {
-    var R = Resource(
-      urlBase + <%-: meta.ctor.getFullPath() | q %>,
+    var lb = function LoopBack(_urlBase) {
+      this.urlBase = _urlBase;
+
+<% for (var modelName in models) {
+     var meta = models[modelName];
+
+     // capitalize the model name
+     modelName = modelName[0].toUpperCase() + modelName.slice(1);
+-%>
+       /**
+        * @ngdoc object
+        * @name <%-: registryName %>.Loopback.<%-: modelName %>
+        * @header <%-: registryName %>.Loopback.<%-: modelName %>
+        * @object
+        *
+        * @description
+        *
+        * A $resource object for interacting with the `<%-: modelName %>` model.
+        *
+        * @param {string} url The URL to use, e.g. `/api` or `//example.com/api`.
+        *
+        * ## Example
+        *
+        * See
+        * {@link http://docs.angularjs.org/api/ngResource.$resource#example $resource}
+        * for an example of using this object.
+        *
+<%        /*
+           * TODO(bajtos) provide an example of performing basic CrUD operations,
+           * including filter/where find query
+           */
+-%>
+       */
+
+      this[<%-: modelName | q %>] = Resource(
```

Comment:
Please save `this[<%-: modelName | q %>]` to `R` and use `R` in the code below to avoid excessive duplication of `this[<%-: modelName | q %>]`. It will keep the diff a bit smaller too.